Slog replaces Result Error type?

I just tried add logging by adding [this slog example](https://github.com/slog-rs/misc/blob/master/examples/file-line-module.rs) to my project. When I compiled it raised an error about a function that returns a result `Result<Mmap, std::io::Error> ` with `expected at most 1 type argument` although the method I call returns a Result like mentioned(well docs say `Result<Mmap>` and usually by default that implies `std::io::Error`. RLS points out that it should be `Result<Mmap, slog::Error> `...?

Perhaps I'm not familiar enough with slog but it seems that it's overriding my error type?

```
error[E0244]: wrong number of type arguments: expected at most 1, found 2
   --> src/main.rs:158:20
    |
158 | fn map_memory() -> Result<Mmap, std::io::Error> {
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected at most 1 type argument
```

Change to one type arg with Result:
```
158 | fn map_memory() -> Result<Mmap> {
    |                    ------------ expected `std::result::Result<memmap::Mmap, slog::Error>` because of return type
159 |     let file = File::open("/dev/shm/looking-glass")?;
160 |     unsafe { memmap::Mmap::map(&file) }
    |              ^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `slog::Error`, found struct `std::io::Error`
    |
    = note: expected type `std::result::Result<_, slog::Error>`
               found type `std::result::Result<_, std::io::Error>`
```

I'm just after logging file, line, and if possible function name. The example seemed to be a good one to try, but I'm a bit surprised that it's affecting other parts of my code like this, presumably something in the example causes this behaviour, can I opt out of it?

--- a/examples/file-line-module.rs
+++ b/examples/file-line-module.rs
@@ -3,7 +3,7 @@ extern crate slog;
 extern crate slog_json;
 extern crate slog_term;
 
-use slog::*;
+use slog::Logger;
 
 fn main() {
     let decorator = slog_term::PlainSyncDecorator::new(std::io::stdout());

